src/extension/src/scripts/generate-test.ts

```diff
@@ -0,0 +1,129 @@
+        payload: { path: CONSTANTS.API.DEVELOPMENT_PATH_TO_TEMPLATES },
+        liveMessageHandler: value => {
+          value;
+        }
+      })
+      .then(() => {
+        instance
+          .getFrameworks(projType)
+          .then(frameworks => {
+            frameworks.forEach(
+              (obj: { tags: { type: string }; name: string }) => {
+                if (obj.tags.type == "frontend") {
+                  frontend.push(obj.name);
+                } else if (obj.tags.type == "backend") {
+                  backend.push(obj.name);
+                }
+              }
+            );
+
+            function getPagesObj(
+              instance: CoreTemplateStudio,
+              projType: string,
+              frontend: any,
+              backend: any
+            ) {
+              instance
+                .getPages(projType, frontend, backend)
+                .then(pages => {
+                  pagesObj = [];
+                  pages.forEach((page: { name: any; templateId: any }) => {
+                    pagesObj.push({
+                      name: page.name,
+                      identity: page.templateId
+                    });
+                  });
+                })
+                .catch(err => {
+                  console.log(err);
+                });
+              return pagesObj;
+            }
+
+            function generateProj(backend: string, frontend: string) {
+              return () => {
+                console.log(`generating ${backend} ${frontend}`);
+                return instance.generate({
+                  port: instance.getPort(),
+                  payload: {
+                    backendFramework: backend,
+                    frontendFramework: frontend,
+                    pages: getPagesObj(instance, projType, frontend, backend),
+                    path: "../../../../../src/extension/src/template_test",
+                    projectName: backend + "-" + frontend,
+                    projectType: projType,
+                    services: []
+                  },
+                  liveMessageHandler: value => {
+                    value;
+                  }
+                });
+              };
+            }
+
+            let prevPromise: Promise<any> = Promise.resolve(null);
+            for (var i = 0; i < frontend.length; i++) {
+              for (var j = 0; j < backend.length; j++) {
+                prevPromise = prevPromise.then(
+                  generateProj(backend[j], frontend[i])
+                );
+              }
+            }
+
+            prevPromise.then(() => {
+              console.log("done");
+            });
+          })
+          .catch((error: Error) => {
+            throw Error(error.toString());
+          });
+      })
+      .catch(() => {
+        syncAttemptNum++;
+        return delay(3000).then(() => attemptSync(instance, syncAttemptNum));
+      });
+  }
+  if (syncAttemptNum >= CONSTANTS.API.MAX_SYNC_REQUEST_ATTEMPTS) {
```

Comment:
NIT: I believe this will be always true, so this condition is redundant, but in general I think it would be nice to move this to the top to reduce nesting:
```suggestion
  if (syncAttemptNum >= CONSTANTS.API.MAX_SYNC_REQUEST_ATTEMPTS) {
    CoreTemplateStudio.DestroyInstance();
    throw new Error("too many failed sync requests");
  }
  return instanceObj...
  //Everything else
```

Reply:
@annielovesjs this is a pretty good suggestion. Have this condition at top to fail. Then you would know that number of sync attempts is less than max allowed for sure and you won't have to wrap that in the `if` clause.

Reply:
actually, this condition is only true when we fail to connect to core and continuously try to sync until we reach a max amount of attempts which generally doesn't happen if we connect to core properly but I did make new changes to wrap that in an else clause instead of having a separate if statement

Reply:
@piotrmark @njindal239 true, good catch! It'll definitely make it a lot cleaner. I'll make the changes now